app/models/mhv_account.rb

```diff
@@ -153,6 +153,7 @@ def previously_registered?
 
   def setup
     raise StandardError, 'You must use find_or_initialize_by(user_uuid: #)' if user_uuid.nil?
+    return if user.nil? # fixes strange flow issue on login / logout because of how user.rb invokes this
```

Comment:
Are we sure this was not due to the UUID with/without dashes issue? Is there an easy way to reproduce the issue that led to this? We can test with my most recent PR and decide if this is needed.